File: common/discord/types/payloads/gateway/events/typing-start.go
package events

import "github.com/discordnova/nova/common/discord/types/structures"

// GatewayEventTypingStartPayload is sent when a user starts typing in a channel.
type GatewayEventTypingStartPayload struct {
	// id of the channel
	ChannelID string `json:"channel_id"`
	// id of the guild
	GuildID string `json:"guild_id,omitempty"`
	// id of the user
	UserID string `json:"user_id"`
	// unix time (in seconds) of when the user started typing
	Timestamp int `json:"timestamp"`
	// the member who started typing if this happened in a guild
	Member structures.GuildMember `json:"member,omitempty"`
}

File: common/rust/src/types/dispatch/typing_start.rs
use crate::types::guild::GuildMember;
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
pub struct TypingStart {
    channel_id: String,
    guild_id: Option<String>,
    user_id: String,
    timestamp: i64,
    member: Option<GuildMember>,
}